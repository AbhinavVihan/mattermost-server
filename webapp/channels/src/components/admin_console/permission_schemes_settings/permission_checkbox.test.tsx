// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import PermissionCheckbox from 'components/admin_console/permission_schemes_settings/permission_checkbox';
import {renderWithIntl} from 'tests/react_testing_utils';

describe('components/admin_console/permission_schemes_settings/permission_checkbox', () => {
    test('should match snapshot on no value', () => {
        const wrapper = renderWithIntl(
            <PermissionCheckbox/>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot on value "checked" and no id', () => {
        const wrapper = renderWithIntl(
            <PermissionCheckbox value='checked'/>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot on value "checked"', () => {
        const wrapper = renderWithIntl(
            <PermissionCheckbox
                value='checked'
                id='uniqId-checked'
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot on value "intermediate"', () => {
        const wrapper = renderWithIntl(
            <PermissionCheckbox
                value='intermediate'
                id='uniqId-checked'
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot on other value', () => {
        const wrapper = renderWithIntl(
            <PermissionCheckbox
                value='other'
                id='uniqId-checked'
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
