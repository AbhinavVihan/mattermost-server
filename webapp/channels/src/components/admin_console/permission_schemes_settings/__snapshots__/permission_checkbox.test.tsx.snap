// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/admin_console/permission_schemes_settings/permission_checkbox should match snapshot on no value 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="permission-check "
        data-testid=""
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="permission-check "
      data-testid=""
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/admin_console/permission_schemes_settings/permission_checkbox should match snapshot on other value 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="permission-check "
        data-testid="uniqId-checked"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="permission-check "
      data-testid="uniqId-checked"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/admin_console/permission_schemes_settings/permission_checkbox should match snapshot on value "checked" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="permission-check checked"
        data-testid="uniqId-checked"
      >
        <svg
          height="18px"
          viewBox="0 0 18 18"
          width="18px"
        >
          <g
            fill-rule="evenodd"
            id="Developer-Export"
            stroke="none"
            stroke-width="1"
          >
            <g
              id="All-Team-Members"
              transform="translate(-285.000000, -933.000000)"
            >
              <g
                transform="translate(245.000000, 698.000000)"
              >
                <g
                  id="Channel-Administrators"
                  transform="translate(0.000000, 106.000000)"
                >
                  <g
                    id="data-table/column-1"
                    transform="translate(11.000000, 86.000000)"
                  >
                    <g
                      id="row-checkbox---Public-Channels"
                      transform="translate(12.000000, 35.000000)"
                    >
                      <g
                        id="Stacked-Group"
                        transform="translate(0.000000, 5.000000)"
                      >
                        <g
                          id="data-table/1st/row-checkbox-label-base"
                        >
                          <g
                            id="check-on-copy-11"
                            transform="translate(14.000000, 0.000000)"
                          >
                            <path
                              d="M19,3 L5,3 C3.9,3 3,3.9 3,5 L3,19 C3,20.1 3.9,21 5,21 L19,21 C20.1,21 21,20.1 21,19 L21,5 C21,3.9 20.1,3 19,3 L19,3 Z M10,17 L5,12.1923077 L6.4,10.8461538 L10,14.3076923 L17.6,7 L19,8.34615385 L10,17 L10,17 Z"
                              id="checkbox---checked"
                            />
                          </g>
                        </g>
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </g>
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="permission-check checked"
      data-testid="uniqId-checked"
    >
      <svg
        height="18px"
        viewBox="0 0 18 18"
        width="18px"
      >
        <g
          fill-rule="evenodd"
          id="Developer-Export"
          stroke="none"
          stroke-width="1"
        >
          <g
            id="All-Team-Members"
            transform="translate(-285.000000, -933.000000)"
          >
            <g
              transform="translate(245.000000, 698.000000)"
            >
              <g
                id="Channel-Administrators"
                transform="translate(0.000000, 106.000000)"
              >
                <g
                  id="data-table/column-1"
                  transform="translate(11.000000, 86.000000)"
                >
                  <g
                    id="row-checkbox---Public-Channels"
                    transform="translate(12.000000, 35.000000)"
                  >
                    <g
                      id="Stacked-Group"
                      transform="translate(0.000000, 5.000000)"
                    >
                      <g
                        id="data-table/1st/row-checkbox-label-base"
                      >
                        <g
                          id="check-on-copy-11"
                          transform="translate(14.000000, 0.000000)"
                        >
                          <path
                            d="M19,3 L5,3 C3.9,3 3,3.9 3,5 L3,19 C3,20.1 3.9,21 5,21 L19,21 C20.1,21 21,20.1 21,19 L21,5 C21,3.9 20.1,3 19,3 L19,3 Z M10,17 L5,12.1923077 L6.4,10.8461538 L10,14.3076923 L17.6,7 L19,8.34615385 L10,17 L10,17 Z"
                            id="checkbox---checked"
                          />
                        </g>
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </g>
        </g>
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/admin_console/permission_schemes_settings/permission_checkbox should match snapshot on value "checked" and no id 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="permission-check checked"
        data-testid=""
      >
        <svg
          height="18px"
          viewBox="0 0 18 18"
          width="18px"
        >
          <g
            fill-rule="evenodd"
            id="Developer-Export"
            stroke="none"
            stroke-width="1"
          >
            <g
              id="All-Team-Members"
              transform="translate(-285.000000, -933.000000)"
            >
              <g
                transform="translate(245.000000, 698.000000)"
              >
                <g
                  id="Channel-Administrators"
                  transform="translate(0.000000, 106.000000)"
                >
                  <g
                    id="data-table/column-1"
                    transform="translate(11.000000, 86.000000)"
                  >
                    <g
                      id="row-checkbox---Public-Channels"
                      transform="translate(12.000000, 35.000000)"
                    >
                      <g
                        id="Stacked-Group"
                        transform="translate(0.000000, 5.000000)"
                      >
                        <g
                          id="data-table/1st/row-checkbox-label-base"
                        >
                          <g
                            id="check-on-copy-11"
                            transform="translate(14.000000, 0.000000)"
                          >
                            <path
                              d="M19,3 L5,3 C3.9,3 3,3.9 3,5 L3,19 C3,20.1 3.9,21 5,21 L19,21 C20.1,21 21,20.1 21,19 L21,5 C21,3.9 20.1,3 19,3 L19,3 Z M10,17 L5,12.1923077 L6.4,10.8461538 L10,14.3076923 L17.6,7 L19,8.34615385 L10,17 L10,17 Z"
                              id="checkbox---checked"
                            />
                          </g>
                        </g>
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </g>
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="permission-check checked"
      data-testid=""
    >
      <svg
        height="18px"
        viewBox="0 0 18 18"
        width="18px"
      >
        <g
          fill-rule="evenodd"
          id="Developer-Export"
          stroke="none"
          stroke-width="1"
        >
          <g
            id="All-Team-Members"
            transform="translate(-285.000000, -933.000000)"
          >
            <g
              transform="translate(245.000000, 698.000000)"
            >
              <g
                id="Channel-Administrators"
                transform="translate(0.000000, 106.000000)"
              >
                <g
                  id="data-table/column-1"
                  transform="translate(11.000000, 86.000000)"
                >
                  <g
                    id="row-checkbox---Public-Channels"
                    transform="translate(12.000000, 35.000000)"
                  >
                    <g
                      id="Stacked-Group"
                      transform="translate(0.000000, 5.000000)"
                    >
                      <g
                        id="data-table/1st/row-checkbox-label-base"
                      >
                        <g
                          id="check-on-copy-11"
                          transform="translate(14.000000, 0.000000)"
                        >
                          <path
                            d="M19,3 L5,3 C3.9,3 3,3.9 3,5 L3,19 C3,20.1 3.9,21 5,21 L19,21 C20.1,21 21,20.1 21,19 L21,5 C21,3.9 20.1,3 19,3 L19,3 Z M10,17 L5,12.1923077 L6.4,10.8461538 L10,14.3076923 L17.6,7 L19,8.34615385 L10,17 L10,17 Z"
                            id="checkbox---checked"
                          />
                        </g>
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </g>
        </g>
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/admin_console/permission_schemes_settings/permission_checkbox should match snapshot on value "intermediate" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="permission-check intermediate"
        data-testid="uniqId-checked"
      >
        <svg
          height="18px"
          viewBox="0 0 18 18"
          width="18px"
        >
          <g
            fill-rule="evenodd"
            id="Developer-Export"
            stroke="none"
            stroke-width="1"
          >
            <g
              id="All-Team-Members"
              transform="translate(-298.000000, -1273.000000)"
            >
              <g
                transform="translate(245.000000, 698.000000)"
              >
                <g
                  id="Channel-Administrators"
                  transform="translate(0.000000, 106.000000)"
                >
                  <g
                    id="data-table/column-1"
                    transform="translate(11.000000, 86.000000)"
                  >
                    <g
                      id="row-checkbox---@all-@channel-Mentions"
                      transform="translate(0.000000, 375.000000)"
                    >
                      <g
                        id="Stacked-Group"
                      >
                        <g
                          id="data-table/1st/row-checkbox-label-base"
                          transform="translate(25.000000, 5.000000)"
                        >
                          <g
                            id="check-on-copy-11"
                            transform="translate(14.000000, 0.000000)"
                          >
                            <path
                              d="M19,3 C20.1,3 21,3.9 21,5 L21,19 C21,20.1 20.1,21 19,21 L5,21 C3.9,21 3,20.1 3,19 L3,5 C3,3.9 3.9,3 5,3 L19,3 Z M7,11 L7,13 L17,13 L17,11 L7,11 Z"
                              id="checkbox---partial"
                            />
                          </g>
                        </g>
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </g>
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="permission-check intermediate"
      data-testid="uniqId-checked"
    >
      <svg
        height="18px"
        viewBox="0 0 18 18"
        width="18px"
      >
        <g
          fill-rule="evenodd"
          id="Developer-Export"
          stroke="none"
          stroke-width="1"
        >
          <g
            id="All-Team-Members"
            transform="translate(-298.000000, -1273.000000)"
          >
            <g
              transform="translate(245.000000, 698.000000)"
            >
              <g
                id="Channel-Administrators"
                transform="translate(0.000000, 106.000000)"
              >
                <g
                  id="data-table/column-1"
                  transform="translate(11.000000, 86.000000)"
                >
                  <g
                    id="row-checkbox---@all-@channel-Mentions"
                    transform="translate(0.000000, 375.000000)"
                  >
                    <g
                      id="Stacked-Group"
                    >
                      <g
                        id="data-table/1st/row-checkbox-label-base"
                        transform="translate(25.000000, 5.000000)"
                      >
                        <g
                          id="check-on-copy-11"
                          transform="translate(14.000000, 0.000000)"
                        >
                          <path
                            d="M19,3 C20.1,3 21,3.9 21,5 L21,19 C21,20.1 20.1,21 19,21 L5,21 C3.9,21 3,20.1 3,19 L3,5 C3,3.9 3.9,3 5,3 L19,3 Z M7,11 L7,13 L17,13 L17,11 L7,11 Z"
                            id="checkbox---partial"
                          />
                        </g>
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </g>
        </g>
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
